cdap-client-tests/src/test/java/io/cdap/cdap/client/MetadataHttpHandlerTestRun.java

```diff
@@ -505,22 +479,12 @@ public void testDeletedProgramHandlerStage() throws Exception {
     Map<String, String> properties = getProperties(program, MetadataScope.USER);
     Assert.assertEquals(2, properties.size());
 
-    // Deploy the app again, but with one less service.
-    appClient.deploy(TEST_NAMESPACE1, createAppJarFile(ConfigurableServiceApp.class),
-                     new ConfigurableServiceApp.ServiceConfig(1));
-
-    // Get properties from deleted service - should return 404
-    Tasks.waitFor(true, () -> getProperties(program, MetadataScope.USER).isEmpty(), 10, TimeUnit.SECONDS);
-
```

Comment:
I changed this test because I believe this is a change in behavior, when deploying new versions of the app it should not affect existing programs. Correct me if I'm wrong @jingtaowu3 @sanjanasandeep

Reply:
so L508 previously was replacing -SNAPSHOT (not creating a new version) and thats why we could do L513 ?

Reply:
The first deployment has two services. this second deployment is deploying a new version of the app with only one service, based on the comment, looks like it is expecting one of the service to be deleted on the second deployment? and so is the metadata for it.  We have not discussed this behavior but apparently we don't remove programs from old pipeline versions